SimplioOfficial/simplio-app
# Task/73 refreshing auhentication (#98)

# Overview

Implementing a http client that can intercept http requests and responses and use our authenticated strategy. Fot that reason we use a http client generator called [Chopper](https://hadrien-lejard.gitbook.io/chopper). We have two http clients. One is public and adds api key to all request. Another is secured that adds a bearer token to the request and implements refreshing strategy via custom Authenticator.

There are implemented following services which handle its endpoint and call methods:

- Sign Up
- Sign In
- Password reset
- Password change
- Refreshing access token (its part of Authenticator)

# Test cases:
- Try to sign in with an existing account. It should login and stay logged after reopening the application
- Try to sign in with a non-existing account. It should not passed. There is no error handeling for UI at this moment.
- Try to reset a password. It should send a email with password reset link. 
- Try to Sign up. Verify your email. And sign in. You should enter the dashboard.
